File: 황재연 (Python)/1주차/프로그래머스_귤고르기.py
from collections import defaultdict

def solution(k, tangerine):
    board = defaultdict(int)
    answer = 0
    for i in tangerine:
        board[i] += 1
        
    for i in sorted(board.values(),reverse=True):
        k -= i
        answer += 1
        if k <= 0:
            break

    return answer


k = 6
tangerine = [1, 3, 2, 5, 4, 5, 2, 3]
result = 3

print(solution(k,tangerine))

File: 백성열 (java)/1주차 문제/프로그래머스_귤고르기.java
import java.util.*;

public class Solution {
    static public int solution(int k, int[] tangerines) {
        int answer = 0;
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int size : tangerines) {
            map.put(size, map.getOrDefault(size, 0) + 1);
        }

        List<Tangerine> ve = new ArrayList<>();
        for (Map.Entry<Integer, Integer> element : map.entrySet()) {
            ve.add(new Tangerine(element.getKey(), element.getValue()));
        }

        Collections.sort(ve, new Comparator<Tangerine>() {
            @Override
            public int compare(Tangerine o1, Tangerine o2) {
                return Integer.compare(o2.num, o1.num);
            }
        });

        int num = 0;
        for (Tangerine temp : ve) {
            if (num < k) {
                num += temp.num;
                answer++;
            } else {
                break;
            }
        }

        return answer;
    }

    static class Tangerine {
        
        int size;
        int num;

        Tangerine(int size, int num) {
            this.size = size;
            this.num = num;
        }
    }
}